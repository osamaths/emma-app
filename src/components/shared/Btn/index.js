import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {colors} from '../../../js/sharedStyle';

const Btn = ({
  title,
  onPress,
  loading,
  disabled,
  style = {},
  titleStyle = {},
  ...rest
}) => {
  return (
    <TouchableOpacity
      style={[styles.container, style]}
      loading={loading}
      disabled={disabled}
      onPress={onPress}
      {...rest}>
      <Text style={[styles.title, titleStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Btn;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary,
    borderRadius: 8,
    height: 40,
    width: '60%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: colors.btnTitle,
    fontWeight: '600',
  },
});
