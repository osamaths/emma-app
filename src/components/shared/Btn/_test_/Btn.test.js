import React from 'react';
import renderer from 'react-test-renderer';
import Btn from '../index';

test('renders correctly', () => {
  const tree = renderer.create(<Btn />).toJSON();
  expect(tree).toMatchSnapshot();
});
