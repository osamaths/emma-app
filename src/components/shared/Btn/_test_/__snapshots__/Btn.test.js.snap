// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4db6ac",
      "borderRadius": 8,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": "60%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontWeight": "600",
        },
        Object {},
      ]
    }
  />
</View>
`;
